### PR TITLE
feat: add retry logic with backoff

### PR DESCRIPTION
**Pull request by orthros**
Detects if an API call to DRGHS returns a non 404 error code (which
indicates the repository is not being tracked in DRGHS) and will retry
with a backoff (up to 5 times with the 5th time being a 16 second wait).

### CODE REVIEW

File: src/issue.ts [hunk 1]
```diff
@@ -82,11 +87,38 @@ async function getRepoIssues(repo: Repo, flags?: Flags): Promise<IssueResult> {
       url = url + `&page=${pageToken}`;
     }
 
-    try {
-      res = await request<IssuesApiResponse>({url});
-    } catch (e) {
-      console.warn(`Error fetching issues for ${repo.repo}.`);
-      // console.warn(e);
+    let tries = 0;
```
Comment (bcoe) [inline src/issue.ts:90]: I would be tempted to pull this in to a little helper module.

File: src/issue.ts [hunk 2]
```diff
@@ -82,11 +87,38 @@ async function getRepoIssues(repo: Repo, flags?: Flags): Promise<IssueResult> {
...
+    while (tries < nBackoff) {
+      try {
+        console.log(tries);
```
Comment (JustinBeckwith) [inline src/issue.ts:93]: Maybe we don't want this :)

File: src/issue.ts [hunk 3]
```diff
@@ -82,11 +87,38 @@ async function getRepoIssues(repo: Repo, flags?: Flags): Promise<IssueResult> {
...
+        // reattempt with backoff
+        if (
+          e.response.status === 404 &&
+          e.response.data.message ===
+            `repository ${repo.repo} is not tracking issues`
+        ) {
+          console.warn(
+            `Repository ${repo.repo} is not tracking issues in DRGHS... skipping.`
+          );
+          return result;
```
Comment (JustinBeckwith) [inline src/issue.ts:107]: This return doesn't seem right.  Is this supposed to short-circuit exit the function?

Comment (orthros) [reply src/issue.ts:106]: That was the idea. If drghs claims it is not tracking that repository, then warn and skip processing that repository all-together.

File: src/issue.ts (hunk 2 above)
Comment (orthros) [reply src/issue.ts]: whoops 😅

**Review by JustinBeckwith — APPROVED** [review]

**Pull request merged**